rclcpp/src/rclcpp/duration.cpp

```diff
@@ -182,6 +182,36 @@ Duration::operator-(const rclcpp::Duration & rhs) const
     rcl_duration_.nanoseconds - rhs.rcl_duration_.nanoseconds);
 }
 
+void
+bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
+{
+  auto abs_dns = (uint64_t)std::abs(dns);
+  auto abs_scale = (uint64_t)std::abs(scale);
```

Comment:
Casting this to be an unsigned integer I think is going to do the wrong thing here. It's used lower as if it's still a float.